packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/PersistedNameTests.kt

```diff
@@ -0,0 +1,312 @@
+        val newSchema = setOf(io.realm.kotlin.entities.migration.after.PersistedNameChangeMigrationSample::class)
+
+        // Open a realm with the old schema
+        val oldConfig = RealmConfiguration
+            .Builder(schema = oldSchema)
+            .name("foo")
+            .directory("$tmpDir/bar")
+            .build()
+        val oldRealm = Realm.open(oldConfig)
+
+        // Add an object to the realm and close it
+        oldRealm.writeBlocking {
+            copyToRealm(io.realm.kotlin.entities.migration.before.PersistedNameChangeMigrationSample())
+        }
+        oldRealm.close()
+
+        // Open a realm with the new schema
+        var migrationTriggered = false
+        val newConfig = RealmConfiguration
+            .Builder(schema = newSchema)
+            .name("foo")
+            .directory("$tmpDir/bar")
+            .schemaVersion(1)
+            .migration(
+                AutomaticSchemaMigration {
+                    migrationTriggered = true
+                }
+            )
+            .build()
+        Realm.open(newConfig).close()
+
+        // Migration should have been triggered
+        assertTrue { migrationTriggered }
+    }
+
+    @Test
+    fun schema_changingPublicName_doesNotTriggerMigration() {
+        val oldSchema = setOf(io.realm.kotlin.entities.migration.before.PublicNameChangeMigrationSample::class)
+        val newSchema = setOf(io.realm.kotlin.entities.migration.after.PublicNameChangeMigrationSample::class)
+
+        // Open a realm with the old schema
+        val oldConfig = RealmConfiguration
+            .Builder(schema = oldSchema)
+            .name("foo.realm")
+            .directory("$tmpDir/bar")
+            .build()
+        val oldRealm = Realm.open(oldConfig)
+
+        // Add an object to the realm and close it
+        oldRealm.writeBlocking {
+            copyToRealm(io.realm.kotlin.entities.migration.before.PublicNameChangeMigrationSample())
+        }
+        oldRealm.close()
+
+        // Open a realm with the new schema
+        val newConfig = RealmConfiguration
+            .Builder(schema = newSchema)
+            .name("foo.realm")
+            .directory("$tmpDir/bar")
+            .migration(
+                AutomaticSchemaMigration {
+                    fail("Migration triggered")
+                }
+            )
+            .build()
+
+        // Migration should not be needed
+        Realm.open(newConfig).close()
+    }
+
+    private fun <T> assertCanQuerySingle(property: KMutableProperty1<PersistedNameSample, T>, nameToQueryBy: String, value: T) {
+        realm.query<PersistedNameSample>("$nameToQueryBy = $0", value)
+            .find()
+            .single()
+            .run {
+                assertNotNull(this)
+                assertEquals(value, property.getValue(this, property))
+            }
+    }
+}
+
+class PersistedNameSample : RealmObject {
```

Comment:
I would try to have 1 or 2 fields stress the unicode support, ie. what happens if we use an emojii as a persisted name? Core should support this.

Reply:
Good idea, will add this.